raiden/transfer/architecture.py

```diff
@@ -138,3 +138,23 @@ def dispatch(self, state_change):
         assert all(isinstance(e, Event) for e in events)
 
         return events
+
+    def __eq__(self, other):
+        if isinstance(other, StateManager):
```

Comment:
It's not really hard to understand, but perhaps overly complicated to overcome the slots problem. What augusto suggests should be fine for state manager equality in all instances.